lib/verifier.py

```diff
@@ -167,6 +167,22 @@ def _raise_if_valid_tx(cls, raw_tx: str):
         tx = Transaction(raw_tx)
         try:
             tx.deserialize()
+            # Besides deserializing, we perfom additional checks to reduce the
+            # chance that a legitimate merkle node trips us up.
+            # A legitimate txn of 64 bytes can only have 1 input and 1 output.
+            txin, = tx.inputs()
+            txout, = tx.outputs()
+            # Txes can be at most 1 MB and and each output requires >= 9 bytes;
+            # Thus prevout_n must be either
+            #   <= 111111 for a real spend, or
+            #   == 0xffffffff for a coinbase.
+            assert txin['prevout_n'] <= 111111 or txin['prevout_n'] == 0xffffffff
+            # Output amount can't possibly be more than 21 million bitcoin.
+            assert txout['value'] < 21e14
```

Comment:
I'm not sure if python floats are infinite precision.... are they?

Might be easier / faster to use ints directly (since Python has bigints) such as:

```
2_100_000_000_000_000
```

Reply:
It does yield a float sadly ---
```
>>> 21e14
2100000000000000.0
```

Reply:
I have been burned a lot by architectures in this situation.  The type you are comparing to is a big int ultimately (or a regular int).  Not that it matters TOO much -- but perhaps for precision's sake care to use.. 2_100_000_000_000_000 which is likely the same data type (64 bit int)?